### PR TITLE
Medical report download

### PR DESCRIPTION
**Pull request by jmnda-dev**
**What is in this PR:**

- [x] Add feature for downloading Beneficiary medical record:

    - Add class for generating a medical report pdf document
    - Add celery task for generating and saving the medical report
    - Add javascript for handling async download of the report
    - A few fixes like a typo in the Beneficiary model field and enable Celery worker service.

**What is remaining?:**
- [x] A few changes need to be made in the class for generating the document.
- [x] The document should be updated to include additional data.
- [x] Style the document to look better.
- [x] Downloading medical report as a zip file which includes supporting documents. 

medical_report_demo

### CODE REVIEW

**Comment by sonlinux** [comment]
Hi @jmnda-dev 
The PR for model changes towards working with just a single model focus is now merged
please pull and fix conflicts and all should be good for you moving forward

File: itez/beneficiary/views.py [hunk 1]
```diff
@@ -195,6 +225,25 @@ class MedicalRecordCreateView(LoginRequiredMixin, CreateView):
...
+
+    def generate_upload_dirname(self, beneficiary_name):
+        uuid_chars = str(uuid.uuid4())[:8]
+        directory_name = f"{beneficiary_name}_{uuid_chars}"
+        return directory_name
+
+    def create_files_dict(self, directory=None, filenames=[]):
+        d = {}
+        d[directory] = filenames
+        return d
```
Comment (sonlinux) [inline itez/beneficiary/views.py]: I would move these 3 methods to utility module as standalone generic reusable functions

File: itez/beneficiary/views.py [hunk 2]
```diff
@@ -50,9 +54,12 @@
     Agent,
     Service,
 )
+from itez.beneficiary.models import Service
+from django.db.models import Count
 
 from django.core.paginator import Paginator
-from .tasks import generate_export_file
+from django.core.paginator import Paginator
+from .tasks import generate_export_file, generate_medical_report
```
Comment (sonlinux) [inline itez/beneficiary/views.py]: lets use a relative import here `itez.beneficiary.tasks import ...`

File: itez/beneficiary/utils.py [hunk 3]
```diff
@@ -1,4 +1,29 @@
...
+    Table,
+    TableStyle,
+    Paragraph,
+)
+from reportlab.pdfbase import pdfmetrics
+from reportlab.pdfbase.ttfonts import TTFont, TTFError
+from reportlab.pdfgen import canvas
+from reportlab.lib import colors
+
+from .resources import BeneficiaryResource
```
Comment (sonlinux) [inline itez/beneficiary/utils.py]: lets use relative import paths here `itez.beneficiary.resources import ...`

File: itez/beneficiary/tasks.py [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
 from django.conf import settings
 
+from celery import shared_task
+
 from config import celery_app
 
+from .models import Beneficiary, MedicalRecord
 from .resources import BeneficiaryResource
-from celery import shared_task
+from .utils import GenerateMedicalReport
```
Comment (sonlinux) [inline itez/beneficiary/tasks.py]: lets use relatve imorts here `itez.beneficiary.utils import ...`

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: itez/beneficiary/views.py (hunk 1 above)
Comment (jmnda-dev) [reply itez/beneficiary/views.py]: That makes sense. I will move it to the utils module.

File: itez/beneficiary/utils.py (hunk 3 above)
Comment (jmnda-dev) [reply itez/beneficiary/utils.py]: I have updated the imports to be relative

File: itez/beneficiary/tasks.py (hunk 4 above)
Comment (jmnda-dev) [reply itez/beneficiary/tasks.py]: I have done the same here as well

**Review by sonlinux — APPROVED** [review]
LGTM

**Pull request merged**